libjolie/src/main/java/jolie/lang/parse/module/SymbolTable.java

```diff
@@ -0,0 +1,180 @@
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301  USA
+ */
+
+package jolie.lang.parse.module;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import jolie.lang.parse.ast.SymbolNode;
+import jolie.lang.parse.ast.SymbolNode.Privacy;
+import jolie.lang.parse.context.ParsingContext;
+import jolie.lang.parse.module.SymbolInfo.Scope;
+import jolie.lang.parse.module.exceptions.DuplicateSymbolException;
+
+/**
+ * A class represent the Symbol table of a Jolie module
+ */
+public class SymbolTable {
+	/**
+	 * Symbol target source
+	 */
+	private final URI source;
+
+	/**
+	 * Maps of Symbolname an corresponding SymbolInfo object
+	 */
+	private final Map< String, SymbolInfo > symbols;
+
+	/**
+	 * A constructor of SymbolTable
+	 * 
+	 * @param source source of Jolie module
+	 */
+	public SymbolTable( URI source ) {
+		this.source = source;
+		this.symbols = new HashMap<>();
+	}
+
+	public URI source() {
+		return this.source;
+	}
+
+	/**
+	 * resolve the wildcard symbol by replace with array of symbol rely in it's symbolTable
+	 * 
+	 * @param wildCardSymbol a wildcard symbol in symbol table to replace
+	 * @param symbolsFromWildcard array of symbols in parsed wildcardSymbol's module
+	 * @throws ModuleException when adding name duplicate name to the symbol
+	 */
+	public void replaceWildCardSymbol( SymbolWildCard wildCardSymbol,
+		SymbolInfo... symbolsFromWildcard ) throws DuplicateSymbolException {
+		for( SymbolInfo symbolFromWildcard : symbolsFromWildcard ) {
+			if( isDuplicateSymbol( symbolFromWildcard.name() ) ) {
+				throw new DuplicateSymbolException( symbolFromWildcard.name() );
+			}
+			if( symbolFromWildcard.privacy() == Privacy.PUBLIC ) {
+				this.symbols.put( symbolFromWildcard.name(), symbolFromWildcard );
+			}
+		}
+		this.symbols.remove( wildCardSymbol.name(), wildCardSymbol );
+	}
+
+	/**
+	 * create and add a local Symbol with it's ASTNode to the table
+	 * 
+	 * @param name Symbol name in local execution context
+	 * @param node an AST node implementing SymbolNode
+	 * 
+	 * @throws ModuleException when adding name duplicate name to the symbol
+	 */
+	public void addSymbol( String name, SymbolNode node ) throws DuplicateSymbolException {
+		if( isDuplicateSymbol( name ) ) {
+			throw new DuplicateSymbolException( name );
+		}
+		this.symbols.put( name, new SymbolInfoLocal( name, node ) );
+	}
+
+	/**
+	 * create and add an external Symbol with it's module target to the table
+	 * 
+	 * @param context Context where symbol is declare
+	 * @param name Symbol name
+	 * @param moduleTargetStrings String array defining target module location defined at import
+	 *        statement
+	 * 
+	 * @throws ModuleException when adding name duplicate name to the symbol
+	 */
+	public void addSymbol( ParsingContext context, String name, List< String > moduleTargetStrings )
```

Comment:
Make a class to represent import paths, called `ImportPath`. Otherwise we'll have to move these lists everywhere and they're not very informative. The name `moduleTargetStrings` is also not very informative, `importPath` will be clearer.